src/v/kafka/server/handlers/join_group.cc

```diff
@@ -40,20 +42,12 @@ ss::future<response_ptr> join_group_handler::handle(
     join_group_request request(ctx);
 
     if (request.data.group_instance_id) {
-        return ctx.respond(
+        co_return co_await ctx.respond(
           join_group_response(error_code::unsupported_version));
     }
 
-    return ss::do_with(
-      std::move(ctx),
-      std::move(request),
-      [](request_context& ctx, join_group_request& request) {
-          return ctx.groups()
-            .join_group(std::move(request))
-            .then([&ctx](join_group_response&& reply) {
-                return ctx.respond(std::move(reply));
-            });
-      });
+    auto resp = co_await ctx.groups().join_group(std::move(request));
+    co_return co_await ctx.respond(std::move(resp));
```

Comment:
wow, so much simpler. i bet raft would benefit from some cleanup too.

Reply:
it would